### PR TITLE
Add function to delete project

### LINKED ISSUE #95: Add a function that deletes a project
We need to expose a function `Dockup.DeleteDeploymentJob.spawn_process(<project_id>, <callback>)` that, given a project id, : 
1. Deletes all containers of the project including any volumes that were created
2. Deletes project_dir of the project
3. Removes nginx configs of the project
4. Triggers callback once done

Also add a service `DockupUi.DeleteDeploymentService` that queues a project deletion using above API.

### PR DESCRIPTION
**Pull request by ganeshmannamal**
resolve #95

### CODE REVIEW

File: apps/dockup/lib/dockup/delete_deployment_job.ex [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  def spawn_process(id, callback) do
+    spawn(fn -> perform(id, callback) end)
+  end
+
+  def perform(project_identifier, callback \\ DefaultCallback.lambda, deps \\ []) do
+    callback.(:deleting_deployment, nil)
+
+    project    = deps[:project]    || Project
```
Comment (emilsoman) [inline apps/dockup/lib/dockup/delete_deployment_job.ex]: This one's find, but going forward, let's not align using `=`. Pollutes git diff when adding or removing lines unnecessarily.

File: apps/dockup_ui/test/services/delete_deployment_service_test.exs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+defmodule DockupUi.DeleteDeploymentServiceTest do
+  use DockupUi.ModelCase, async: true
+  use Timex
+  import DockupUi.Factory
+
+  alias DockupUi.DeleteDeploymentService
+
+  defmodule FakeDeleteDeploymentJob do
+    def spawn_process(_id, _callback) do
```
Comment (emilsoman) [inline apps/dockup_ui/test/services/delete_deployment_service_test.exs]: Maybe replace `_id` with 1 so we know the correct deployment id is passed to `spawn_process`

File: apps/dockup/lib/dockup/container.ex [hunk 3]
```diff
@@ -104,7 +104,7 @@ defmodule Dockup.Container do
   end
 
   def stop_containers(project_id, command \\ Dockup.Command) do
-    Logger.info "Stopping containers of project #{project_id}"
+    Logger.info "Stopping deployment-#{project_id} containers"
     command.run("docker-compose", ["-p", "#{project_id}", "stop"], Dockup.Project.project_dir(project_id))
```
Comment (emilsoman) [inline apps/dockup/lib/dockup/container.ex]: Instead of `docker-compose stop`, let's use `docker-compose -p <project_id> down -v`

**Comment by ganeshmannamal** [comment]
@emilsoman pushed the fixes requested

**Pull request merged**